### PR TITLE
vue-assignment-v1: Onboarding website.

### PR DESCRIPTION
**Pull request by AyushChaudhary2002**

### CODE REVIEW

File: onboarding-website/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  height: 79vh;
+  margin: 1vh 0vh;
+}
+.aside{
+  width: 20vw;
+  margin:1rem;
+  padding: 0.5rem;
+}
+.section{
+  width:70vw;
```
Comment (mayankcompro) [inline onboarding-website/src/App.vue]: space after `:`

File: onboarding-website/src/App.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+.header{
+  width: 100vw;
+  background-color: #673ab5;
+}
+.main{
+  display: flex;
+  height: 79vh;
+  margin: 1vh 0vh;
+}
+.aside{
```
Comment (mayankcompro) [inline onboarding-website/src/App.vue]: empty line space above every css class declaration.

File: onboarding-website/src/components/asideComponent.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        single-line />
+        <template v-slot:prepend>
+        <v-avatar class="me-4 mt-2" rounded="0">
+        <img src="/src/assets/magnify.svg" alt="" srcset="">
+        </v-avatar>
+        </template>
+    </v-toolbar>
+
+  
+
```
Comment (mayankcompro) [inline onboarding-website/src/components/asideComponent.vue]: Too many extra empty lines.

File: onboarding-website/src/components/asideComponent.vue [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    <v-list>
+      <v-list-item v-for="(item, i) in searching" :key="i" link>
```
Comment (mayankcompro) [inline onboarding-website/src/components/asideComponent.vue]: What is `i`? In quantitative programming, we usually have a practice to use `i` as a variable, whenever you write any code while developing any application use a quality name and descriptive too.

File: onboarding-website/src/components/asideComponent.vue [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        v-model="data.search"
+        clearable
+        hide-details
+        label="Search Topics"
+        single-line />
+        <template v-slot:prepend>
+        <v-avatar class="me-4 mt-2" rounded="0">
+        <img src="/src/assets/magnify.svg" alt="" srcset="">
+        </v-avatar>
+        </template>
```
Comment (mayankcompro) [inline onboarding-website/src/components/asideComponent.vue]: indentation issue.

File: onboarding-website/src/components/asideComponent.vue [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        image: '/src/assets/javaScript.svg',
+        category: 'Javascript'
+      },
+      {
+        image: '/src/assets/vue.svg',
+        category: 'Vue.Js'
+      }
+    ],
+    search: ''
+  })
```
Comment (mayankcompro) [inline onboarding-website/src/components/asideComponent.vue]: missing semi-colon.

File: onboarding-website/src/components/asideComponent.vue [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        <v-divider></v-divider>
+      </v-list-item>
+    </v-list>
+  </v-card>
+</template>
+
+<script setup>
+import { reactive, computed } from 'vue';
+const data = reactive({
+    items: [
```
Comment (mayankcompro) [inline onboarding-website/src/components/asideComponent.vue]: indentation issue.

File: onboarding-website/src/components/asideComponent.vue [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        category: 'Vue.Js'
+      }
+    ],
+    search: ''
+  })
+
+ const searching = computed(() => {
+    if (!data.search) return data.items
+
+    const search = data.search.toLowerCase()
```
Comment (mayankcompro) [inline onboarding-website/src/components/asideComponent.vue]: Lot of missing semi-colon.

File: onboarding-website/src/components/navbarComponent.vue [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        <a class="link" href="#">About</a>
+        <a class="link" href="#">Contact</a>
+      </div>
+      <v-card
+            class="mb-2"
+            density="compact"
+            prepend-avatar="/src/assets/admin.svg"
+            subtitle="Instructor"
+            title= "Admin"
+            variant="text"/>
```
Comment (mayankcompro) [inline onboarding-website/src/components/navbarComponent.vue]: indentation issue.

**Review by mayankcompro — CHANGES_REQUESTED** [review]
Improve all indentation and other silly bad practices.

File: onboarding-website/src/components/asideComponent.vue (hunk 8 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/components/asideComponent.vue]: resolve

File: onboarding-website/src/App.vue (hunk 1 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/App.vue]: resolve

File: onboarding-website/src/App.vue (hunk 2 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/App.vue]: resolve

File: onboarding-website/src/components/asideComponent.vue (hunk 3 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/components/asideComponent.vue]: resolve

File: onboarding-website/src/components/asideComponent.vue (hunk 4 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/components/asideComponent.vue]: resolve

File: onboarding-website/src/components/asideComponent.vue (hunk 5 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/components/asideComponent.vue]: resolve

File: onboarding-website/src/components/asideComponent.vue (hunk 6 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/components/asideComponent.vue]: resolve

File: onboarding-website/src/components/asideComponent.vue (hunk 7 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/components/asideComponent.vue]: resolve

File: onboarding-website/src/components/navbarComponent.vue (hunk 9 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/components/navbarComponent.vue]: resolve

File: onboarding-website/src/App.vue [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    </section>
+  </main>
+</template>
+
+<script setup>
+  import navbarComponent from '@/components/navbarComponent.vue';
+  import asideComponent from '@/components/asideComponent.vue';
+  import sectionComponent from './components/sectionComponent.vue';
+  import { useTaskStore } from './stores/taskData';
+  import { useCategoryStore } from './stores/category';
```
Comment (PurnimaCompro) [inline onboarding-website/src/App.vue]: alphabetical order

File: onboarding-website/src/App.vue [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  
+  import { ref } from 'vue';
+  
+  const filterCriteria = ref('');
+  function filter(value) {
+    filterCriteria.value = value.toUpperCase();
+  }
+
```
Comment (PurnimaCompro) [inline onboarding-website/src/App.vue]: User Es6 syntax

File: onboarding-website/src/main.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
+import { createApp } from 'vue';
+import { createPinia } from 'pinia';
+
```
Comment (mayankcompro) [inline onboarding-website/src/main.js]: Why empty new line here and below?

File: onboarding-website/src/App.vue [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+  import { useCategoryStore } from './stores/category';
+  
+  import { ref } from 'vue';
+  
+  const filterCriteria = ref('');
+  function filter(value) {
+    filterCriteria.value = value.toUpperCase();
+  }
+  
+  if (localStorage.getItem('taskStorage') && useTaskStore().data.task.length === 0) {
```
Comment (PurnimaCompro) [inline onboarding-website/src/App.vue]: I guess comparison with 0 is not needed.

File: onboarding-website/src/router/index.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
+import { createRouter, createWebHistory } from 'vue-router'
+import HomeView from '../views/HomeView.vue'
```
Comment (mayankcompro) [inline onboarding-website/src/router/index.js]: Semi colon missing.

**Review by mayankcompro — CHANGES_REQUESTED** [review]

File: onboarding-website/src/components/modalComponent.vue [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+<script setup>
+  import { ref, reactive } from 'vue';
+  import shortId from 'short-uuid';
+  import { useTaskStore } from '@/stores/taskData.js';
+  import { useCategoryStore } from '../stores/category';
+
+  const props = defineProps({
+    data: {
+      type: Object
+    }
```
Comment (PurnimaCompro) [inline onboarding-website/src/components/modalComponent.vue]: in one line

**Review by PurnimaCompro — CHANGES_REQUESTED** [review]

File: onboarding-website/src/App.vue [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+    const tempData = JSON.parse(localStorage.getItem('taskStorage'));
+    tempData.forEach((ele) => {
+      useTaskStore().addTask(ele);
+    });
+  }
+  
+  if (localStorage.getItem('categoryStorage') && useCategoryStore().data.items.length === 0) {
```
Comment (mayankcompro) [inline onboarding-website/src/App.vue]: This line can be written as -
```
if (localStorage.getItem('categoryStorage') && !useCategoryStore().data.items.length) {
```
Please validate and implement.

**Review by mayankcompro — CHANGES_REQUESTED** [review]

File: onboarding-website/src/App.vue [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+    });
+  }
+</script>
+  
+<style scoped>
+  * {
+      box-sizing: border-box;
+      font-family: sans-serif;
+      margin: 0;
+      padding: 0;
```
Comment (mayankcompro) [inline onboarding-website/src/App.vue]: Indentation issue.

File: onboarding-website/src/stores/category.js [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+      if (item.category !== 'all') temp.push(item.category.toUpperCase());
+    });
+    return temp;
+  });
+
+  const addCategory = (newCategory) => {
+    data.items.push(newCategory);
+  };
+
+  return { data, category, addCategory };
```
Comment (mayankcompro) [inline onboarding-website/src/stores/category.js]: Arrange these in alphabetical order and at other relevant pages.

**Review by mayankcompro — CHANGES_REQUESTED** [review]

File: onboarding-website/src/stores/category.js (hunk 18 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/stores/category.js]: resolve

File: onboarding-website/src/App.vue (hunk 16 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/App.vue]: resolve

File: onboarding-website/src/App.vue (hunk 13 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/App.vue]: resolve

File: onboarding-website/src/App.vue (hunk 11 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/App.vue]: resolve

File: onboarding-website/src/App.vue (hunk 10 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/App.vue]: resolve

File: onboarding-website/src/router/index.js (hunk 14 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/router/index.js]: resolve

File: onboarding-website/src/main.js (hunk 12 above)
Comment (AyushChaudhary2002) [reply onboarding-website/src/main.js]: resolve

**Pull request merged**